src/components/Authentication/Authentication.vue

```diff
@@ -0,0 +1,94 @@
+<template lang="pug">
+v-flex.pa-2.girder-authentication-component(xs-12)
+  v-card
+    v-tabs(v-model="activeTab", color="primary", slider-color="yellow", dark)
+      v-tab(key="login") Log In
+      v-tab(key="registration", v-if="register") Register
+      v-tab-item(key="login-box")
+        login(ref="loginForm",
+            :oauth-providers="oauthProviders",
+            :forgot-password-url="forgotPasswordUrl",
+            :forgot-password-route="forgotPasswordRoute",
+            @forgotpassword="$emit('forgotpassword')")
+      v-tab-item(key="registration-box", v-if="register")
+        registration(
+            ref="registerForm",
+            :oauth-providers="oauthProviders")
+</template>
+
+<script>
+import Login from './Login.vue';
+import Registration from './Register.vue';
+
+export default {
+  inject: ['girderRest'],
+  props: {
+    register: {
+      type: Boolean,
+      default: false,
+      note: 'Enable registration from this component?',
+    },
+    oauth: {
+      type: Boolean,
+      default: false,
+      note: 'Enable OAuth login from this component?',
+    },
+    forgotPasswordUrl: {
+      type: String,
+      note: 'A full URL to be used as an anchor href to an external page.  Overrides forgotPasswordRoute.',
+    },
+    forgotPasswordRoute: {
+      type: Object,
+      note: 'A vue-router route path.  Must use object form.',
+    },
+  },
+  components: {
+    Login,
+    Registration,
+  },
+  data() {
+    return {
+      activeTab: null,
+      loginInProgress: false,
+      registerInProgress: false,
+      loginErrors: null,
+      registerErrors: null,
+      oauthProviders: [],
+    };
+  },
+  async mounted() {
+    if (this.oauth) {
+      const resp = await this.girderRest.fetchOauth();
+      this.oauthProviders = [];
+      resp.data.forEach((provider) => {
+        this.oauthProviders.push({
+          id: provider.id,
+          name: provider.name,
+          url: provider.url,
+        });
+      });
+    }
+  },
+};
+</script>
+
+<style lang="stylus">
```

Comment:
I intentionally made these styles without `scoped` because they're used in all the children -- OAuth, Register, Login, etc.

I didn't see a policy for duplicate CSS in this project, so I named a class after this module to manage the scope.  I saw this policy implemented in [resonantgeo](https://github.com/Kitware/resonantgeo/blob/b5665b998de0de6731546aa2302d9d182f16d0db/src/components/AppToolbar.vue#L45).

We can totally duplicate these styles into the other vue components, but it may be more difficult to keep styles consistent if you have to check all children when styles are changed.  I'd like to get @jbeezley's thoughts here as well.

Reply:
That's done in resonantgeo because it alters the CSS of the child component `.v-toolbar__content` for which there is no other way.

I haven't though about *sharing* CSS between components.  This would work, but would implicitly assume that the child components are not used with out the parent.  A stackoverflow answer mentions [css modules](https://vue-loader.vuejs.org/guide/css-modules.html) as a possible strategy, but I don't really know anything about them.  Another possibility is a `.styl` file that is imported by all of the components as
```html
<style lang="stylus" src="./common.styl" scoped></style>
```